cpp/src/reader/tsfile_reader.cc

```diff
@@ -37,29 +54,74 @@ TsFileReader::~TsFileReader() {
         delete read_file_;
         read_file_ = nullptr;
     }
+    return ret;  // TO DO
 }
 
-int TsFileReader::open(const std::string &file_path) {
+int TsFileReader::query(QueryExpression *qe, ResultSet *&ret_qds) {
+    return tsfile_executor_->execute(qe, ret_qds);
+}
+
+int TsFileReader::query(std::vector<std::string> &path_list, int64_t start_time,
+                        int64_t end_time, ResultSet *&result_set) {
     int ret = E_OK;
-    read_file_ = new storage::ReadFile;
-    tsfile_executor_ = new storage::TsFileExecutor();
-    if (RET_FAIL(read_file_->open(file_path))) {
-        std::cout << "filed to open file " << ret << std::endl;
-    } else if (RET_FAIL(tsfile_executor_->init(read_file_))) {
-        std::cout << "filed to init " << ret << std::endl;
+    Filter *time_filter = new TimeBetween(start_time, end_time, false);
+    Expression *exp =
+        new storage::Expression(storage::GLOBALTIME_EXPR, time_filter);
+    std::vector<Path> path_list_vec;
+    for (const auto &path : path_list) {
+        uint32_t last_point_pos = path.find_last_of('.');
+        if (last_point_pos <= 0) {
+            return E_INVALID_PATH;
+        }
+        std::string device_name = path.substr(0, last_point_pos);
+        std::string measurement_name =
+            path.substr(last_point_pos + 1, path.size() - last_point_pos);
+        path_list_vec.emplace_back(Path(device_name, measurement_name));
```

Comment:
This may not be so simple. A path segment may contain "." when surrounded by "\`".
For example, a path may be "root.db1.d1.\`s1.1\`". 
In this case, this split is invalid.